### PR TITLE
checkEnvironment Precompile method

### PR DESCRIPTION
**Pull request by SergioDemianLerner**

### CODE REVIEW

File: IPs/RSKIP208.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+|**Status**     |Draft |
+
+
+# **Abstract**
+
+This RSKIP proposes to add a method `checkEnvironment()` to a new `Environment`  pre-compile contract. The new method receives as argument a list of fields that represents the minimum environment requirements to execute a method and it checks that the requirements are satisfied. This RSKIP provides the same functionality of [RKIP203](https://github.com/rsksmart/RSKIPs/blob/master/IPs/RSKIP203.md), but in a way which allows the future addition of additional environment variables, such as rent gas limit, for storage rent accounting.
+
+## Motivation
+
+The motivation is the one described in [RSKIP203](https://github.com/rsksmart/RSKIPs/blob/master/IPs/RSKIP203.md) plus we add the requirement that contracts that forward contracts do not become vulnerable if another counter of any kind is added to the environment.
```
Comment (diegomasini) [inline IPs/RSKIP208.md:21]: Could "contracts that forward contracts" be replaced by "contracts that forward contract calls" for clarity?

File: IPs/RSKIP208.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+This RSKIP proposes to create a new `Environment` pre-compile contract to with a method to check if the minimal environment requirements given as the `env` argument are satisfied. The argument contains a list of fields in RLP format. This RSKIP defines that the first argument of the list is the stack depth. The gas limit is left outside `env` because the caller must provide the value externally to `CALL`. The remaining arguments are undefined and are ignored. Users and relayers will negotiate a content for `env`. Smart contract do not need to interpret the content of `env`, and can simply pass it to the `checkEnvironment()` method.
```
Comment (diegomasini) [inline IPs/RSKIP208.md:23]: Remove **to** from "pre-compile contract to with a "

File: IPs/RSKIP208.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+## Specification
+
+The method with signature ` checkEnvironment(bytes env) returns (uint32)` is added to a new pre-compiled contract named `Environment`, created at address `0x0000000000000000000000000000000001000011`. The call returns 0 on success. On failure it returns the number of item in `env` that is not satisfied by the environment (1==first item).
```
Comment (diegomasini) [inline IPs/RSKIP208.md:27]: Replace "failure it returns the number of item in" for "failure it returns the **position** of the item in"

**Review by diegomasini — COMMENTED** [review]
Please accept or reject the proposed modifications so I can merge this PR.

File: IPs/RSKIP208.md (hunk 1 above)
Comment (SergioDemianLerner) [reply IPs/RSKIP208.md:21]: ok

File: IPs/RSKIP208.md (hunk 3 above)
Comment (SergioDemianLerner) [reply IPs/RSKIP208.md]: ok

File: IPs/RSKIP208.md (hunk 2 above)
Comment (SergioDemianLerner) [reply IPs/RSKIP208.md]: ok

**Review by diegomasini — APPROVED** [review]
LGTM!

**Pull request merged**